lib/perl/Genome/ConfigSpec.pm

```diff
@@ -0,0 +1,54 @@
+package Genome::ConfigSpec;
+
+use strict;
+use warnings;
+
+use Mouse;
+
+use Genome::Carp qw(croakf);
+use Module::Runtime qw();
+use Path::Class qw();
+use Try::Tiny qw(try catch);
+use YAML::Syck qw();
+
+has 'type' => (is => 'ro', isa => 'Str', required => 1);
+has 'key'  => (is => 'ro', isa => 'Str', required => 1);
+
+has 'env'           => (is => 'ro', isa => 'Str', predicate => 'has_env');
+has 'default_value' => (is => 'ro', isa => 'Str', predicate => 'has_default_value');
+has 'validators'    => (is => 'ro', isa => 'ArrayRef[CodeRef]', default => sub { [] });
+
+sub new_from_file {
+    my $class = shift;
+    my $file = Path::Class::File->new(shift);
+
+    (my $key = $file->basename) =~ s/\.yaml$//i;
```

Comment:
You can pass fileparse a suffix list. Presumably you are only looking for '.YAML' and '.yaml'.